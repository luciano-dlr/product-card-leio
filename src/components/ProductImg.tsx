import React from 'react';
import { CSSProperties, useContext } from "react";
import NoImage from "../assets/no-image.jpg";
import styles from "../styles/styles.module.css";
import ProductContext from "./ProductCardContext";


export interface ProductCardImgProps {
  className?: string;
  style?: CSSProperties;
  img?: string;
}

export const ProductImg = ({ className, style, img }: ProductCardImgProps) => {
  const { product } = useContext(ProductContext);
  // const { img } = product;


  let imgToShow: string;

  if (img) {
    imgToShow = img;
  } else if (product.img) {
    imgToShow = product.img
  } else {
    imgToShow = NoImage;
  }

  return (
    <img
      className={`${styles.productImg} ${className}`}
      src={imgToShow}
      style={style}
      alt="Product"
    />
  );
};
