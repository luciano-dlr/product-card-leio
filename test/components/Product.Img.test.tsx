import React from "react"
import renderer from "react-test-renderer"
import { ProductImg, ProductCard } from '../../src/components/';
import { product2 } from '../data/product';


describe('Product title and Card', () => {

    test('Debe de mostrar el componente correctamente con Imagen Correspondiente', () => {

        const wraper = renderer.create(
            <ProductImg img="https://hola.jpg" />
        )

        expect(wraper.toJSON()).toMatchSnapshot();


    })
    test('Debe de mostrar el componente correctamente con Imagen Correspondiente', () => {

        const wraper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <>
                            <ProductImg />
                        </>
                    )
                }

            </ProductCard>
        )
        expect(wraper.toJSON()).toMatchSnapshot();
    })
})