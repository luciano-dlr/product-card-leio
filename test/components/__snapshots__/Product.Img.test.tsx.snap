// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product title and Card Debe de mostrar el componente correctamente con Imagen Correspondiente 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;

exports[`Product title and Card Debe de mostrar el componente correctamente con Imagen Correspondiente 2`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
